### PR TITLE
Refactor optimization unit test. Describe the case in xml file,…

### PR DESCRIPTION
**Pull request by boyjoy1127**
Refactor optimization unit test. Describe the case in xml file, and use junit parameterized method to test.

Ref #19937 .

Changes proposed in this pull request:
  - Refactor optimization unit test. Describe the case in xml file, and use junit parameterized method to test.

---

Before committing this PR, I'm sure that I have checked the following options:
- [X] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [X] I have self-reviewed the commit code.
- [X] I have (or in comment I request) added corresponding labels for the pull request.
- [X] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [X] I have made corresponding changes to the documentation.
- [X] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Review by strongduanmu — CHANGES_REQUESTED** [review]
Hi @boyjoy1127, thank you for your contribution. Can you fix ci error first?

**Comment by boyjoy1127** [comment]
> Hi @boyjoy1127, thank you for your contribution. Can you fix ci error first?

Hi, I've fix the bug. @strongduanmu

File: kernel/sql-federation/optimizer/src/test/java/org/apache/shardingsphere/sqlfederation/optimizer/common/TestCaseAssertion.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+
+/**
+ * JAXB definition of test case assertion.
+ */
+@Getter
+@Setter
+@XmlAccessorType(XmlAccessType.FIELD)
+public class TestCaseAssertion {
```
Comment (strongduanmu) [inline kernel/sql-federation/optimizer/src/test/java/org/apache/shardingsphere/sqlfederation/optimizer/common/TestCaseAssertion.java:33]: Please add final for TestCaseAssertion.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: kernel/sql-federation/optimizer/src/test/java/org/apache/shardingsphere/sqlfederation/optimizer/common/TestCaseAssertion.java (hunk 1 above)
Comment (boyjoy1127) [reply kernel/sql-federation/optimizer/src/test/java/org/apache/shardingsphere/sqlfederation/optimizer/common/TestCaseAssertion.java]: I've fixed the this.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**